Tasks/NuGetRestoreV1/nugetinstaller.ts

```diff
@@ -232,9 +232,15 @@ async function main(): Promise<void> {
             credCleanup();
         }
 
+        if (!tl.getVariable('PASS_DEPRECATED_TASK')) {
+            throw new Error(tl.loc("DeprecatedTask"));
+        }
```

Comment:
This would be safer: (but please test)

tl.getVariable('PASS_DEPRECATED_TASK').toLowerCase() === "true"

getVariable may not return the right variable type for this; ! on a string might not return the expected result

Reply:
Changed this to if(tl.getVariable('FAIL_DEPRECATED_TASK'))

Reply:
Changed to:  

let shouldFail = tl.getVariable('FAIL_DEPRECATED_TASK');
if (shouldFail != null && shouldFail.toLowerCase() === 'true')

Reply:
Please use `if(tl.getBoolFeatureFlag('FAIL_DEPRECATED_TASK'))` instead.
https://github.com/microsoft/azure-pipelines-task-lib/blob/5ab4ebe0c1c1472e62e08fd10001983ed5d01e40/node/task.ts#L285